lintipy.py

```diff
@@ -217,44 +219,56 @@ def run_process(self, code_path):
-        info = resource.getrusage(resource.RUSAGE_CHILDREN)
-        log = process.stdout.read()
-        logger.debug(log)
-        logger.debug('exit %s', process.returncode)
-        logger.info('Saving log to S3')
-        key = os.path.join(self.label, self.full_name, "%s.log" % self.sha)
-        self.s3.put_object(
-            ACL='public-read',
-            Bucket=self.bucket,
-            Key=key,
-            Body=log,
-            ContentType='text/plain'
-        )
-        logger.info(
-            'linter exited with status code %s in %ss' % (process.returncode, info.ru_utime)
-        )
-        return (
-            info.ru_utime,
-            process.returncode,
-            "https://{0}.s3.amazonaws.com/{1}".format(self.bucket, key),
-        )
+        try:
+            process = subprocess.run(
+                ('python', '-m', self.cmd) + self.cmd_args,
+                stdout=subprocess.PIPE, stderr=subprocess.STDOUT,
+                cwd=code_path, env=self.get_env(),
+                timeout=self.cmd_timeout,
+            )
+        except subprocess.TimeoutExpired:
+            self.set_status(ERROR, 'Command timed out after %ss' % self.cmd_timeout,
+                            'https://lambdalint.github.io/#faq')
+            raise
+        else:
+            info = resource.getrusage(resource.RUSAGE_CHILDREN)
+            log = process.stdout
+            logger.debug(log)
+            logger.debug('exit %s', process.returncode)
+            logger.info('Saving log to S3')
+            key = os.path.join(self.label, self.full_name, "%s.log" % self.sha)
+            self.s3.put_object(
+                ACL='public-read',
+                Bucket=self.bucket,
+                Key=key,
+                Body=log,
+                ContentType='text/plain'
+            )
+            logger.info(
+                'linter exited with status code %s in %ss' % (process.returncode, info.ru_utime)
+            )
+            return (
+                info.ru_utime,
+                process.returncode,
+                "https://{0}.s3.amazonaws.com/{1}".format(self.bucket, key),
+            )
 
     def download_code(self):
         """Download code to local filesystem storage."""
         logger.info('Downloading: %s', self.archive_url)
-        response = self.session.get(self.archive_url)
-        response.raise_for_status()
-        with BytesIO() as bs:
-            bs.write(response.content)
-            bs.seek(0)
-            path = tempfile.mkdtemp()
-            with tarfile.open(fileobj=bs, mode='r:gz') as fs:
-                fs.extractall(path)
-            folder = os.listdir(path)[0]
-            return os.path.join(path, folder)
+        try:
+            response = self.session.get(self.archive_url, timeout=30)
+        except requests.Timeout:
+            self.set_status(ERROR, 'Downloading code timed out after %ss' % self.download_timeout,
+                            'https://lambdalint.github.io/#faq')
```

Comment:
Since it is being used twice, maybe we should move it to a constant or a setting? So we would not forget to update all the places if needed.